### PR TITLE
Add timeout argument to Devnet

### PR DESCRIPTION
**Pull request by Nathan-SL**
## Usage related changes

- Timeout to Gunicorn server can be set using the flag `--timeout` or `-t`
- Ex. `starknet-devnet --timeout 120`
- Resolves issue `Error: ServerDisconnectedError: Server disconnected` when deploying bigger smart contracts.
## Development related changes

- NA

## Checklist:

- [x] Applied formatting - `./scripts/format.sh`
- [x] No linter errors - `./scripts/lint.sh`
- [x] Performed code self-review
- [x] Rebased to the last commit of the target branch (or merged it into my branch)
- [x] Documented the changes
- [ ] Linked the issues which this PR resolves
- [x] All tests are passing

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -427,6 +427,14 @@ Devnet can be started with the `--start-time` argument.
 starknet-devnet --start-time START_TIME_IN_SECONDS
 ```
 
+### Timeout
+
+Timeout can be passed to Devnet's HTTP server.
```
Comment (FabijanC) [inline README.md:432]: Describe here in more detail why it could be useful.

File: starknet_devnet/devnet_config.py [hunk 2]
```diff
@@ -140,6 +141,13 @@ def parse_args(raw_args: List[str]):
         help="Specify the gas price in wei per gas unit; "
         + f"defaults to {DEFAULT_GAS_PRICE:g}",
     )
+    parser.add_argument(
+        "--timeout",
+        "-t",
+        action=NonNegativeAction,
+        default=DEFAULT_TIMEOUT,
+        help="Specify timeout for the server"
```
Comment (FabijanC) [inline starknet_devnet/devnet_config.py:149]: State the default value in the help string (see other arg params for reference)

**Review by FabijanC — CHANGES_REQUESTED** [review]
Great! Has this approach been tested with the user who originally had the problem?

File: starknet_devnet/constants.py [hunk 3]
```diff
@@ -19,3 +19,4 @@
 DEFAULT_ACCOUNTS = 10
 DEFAULT_INITIAL_BALANCE = 10**21
 DEFAULT_GAS_PRICE = 10**11
+DEFAULT_TIMEOUT = 60
```
Comment (FabijanC) [inline starknet_devnet/constants.py:22]: Add comment about it being seconds (or change var name)

File: README.md (hunk 1 above)
Comment (FabijanC) [reply README.md:432]: Also, somewhere in the readme there is a list of all CLI params, that should also be updated

**Review by FabijanC — CHANGES_REQUESTED** [review]

File: starknet_devnet/constants.py (hunk 3 above)
Comment (Nathan-SL) [reply starknet_devnet/constants.py]: https://github.com/Shard-Labs/starknet-devnet/pull/256/commits/b715d1628e248a90cc36c49fd6aa78c9d117f1fb

File: starknet_devnet/devnet_config.py (hunk 2 above)
Comment (Nathan-SL) [reply starknet_devnet/devnet_config.py]: https://github.com/Shard-Labs/starknet-devnet/pull/256/commits/b715d1628e248a90cc36c49fd6aa78c9d117f1fb

File: README.md (hunk 1 above)
Comment (Nathan-SL) [reply README.md]: https://github.com/Shard-Labs/starknet-devnet/pull/256/commits/b715d1628e248a90cc36c49fd6aa78c9d117f1fb

File: README.md [hunk 4]
```diff
@@ -90,6 +90,8 @@ optional arguments:
   --gas-price GAS_PRICE, -g GAS_PRICE
                         Specify the gas price in wei per gas unit; defaults to
                         1e+11
+  --timeout TIMEOUT, -t TIMEOUT
+                        Specify the timout for devnet server; defaults to 60 seconds
```
Comment (FabijanC) [inline README.md:94]: timout -> timeout

Comment (Nathan-SL) [reply README.md]: Great catch! https://github.com/Shard-Labs/starknet-devnet/pull/256/commits/2bca8acd6d94c7939fa47c24c87fcb09466e31e6

**Pull request merged**